main/util.py

```diff
@@ -214,16 +216,38 @@ def create_name_from_email(email):
   return re.sub(r'_+|-+|\.+|\++', ' ', email.split('@')[0]).title()
 
 
-def password_hash(user_db, password):
+def password_hash(password, salt):
   m = hashlib.sha256()
-  m.update(user_db.key.urlsafe())
-  m.update(user_db.created.isoformat())
-  m.update(m.hexdigest())
   m.update(password.encode('utf-8'))
-  m.update(config.CONFIG_DB.salt)
+  m.update(salt)
   return m.hexdigest()
 
 
+def password_salt(n_bytes=32):
+  return binascii.hexlify(os.urandom(n_bytes))
+
+
+# Taken from Django Source Code
+# Helps prevent timing attacks
+# See http://carlos.bueno.org/2011/10/timing.html
+def constant_time_compare(val1, val2):
```

Comment:
@lipis This function secures against a [Timing Attack](https://en.wikipedia.org/wiki/Timing_attack). In python, == returns false as soon as there is a single difference in the values. Thus, an attacker could theoretically [crack the password one letter at a time](http://codahale.com/a-lesson-in-timing-attacks/). This function always takes the same amount of time to finish, regardless of whether the two strings match or not, which is why Django uses it.

Even if you don't accept this whole PR, it might be worth adding this in.